### PR TITLE
Update analyzer tests to JUnit

### PR DESCRIPTION
**Pull request by Damans227**
Linked to https://github.com/trinodb/trino/issues/9378

- Build was successful after making the changes
- All tests passed locally in the IDE

### CODE REVIEW

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java [hunk 1]
```diff
@@ -16,14 +16,15 @@
...
 import java.util.Optional;
 
 import static io.trino.spi.type.BigintType.BIGINT;
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertFalse;
-import static org.testng.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java:27]: Remove a redundant line.

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java [hunk 2]
```diff
@@ -16,14 +16,15 @@
...
 
 import java.util.Optional;
 
 import static io.trino.spi.type.BigintType.BIGINT;
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertFalse;
-import static org.testng.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java:26]: Replace the order of `assertTrue` and `assertFalse`.

**Review by ebyhr — COMMENTED** [review]
Could you setup codestyle in your IDE? This is the documentation. 
https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#code-style

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestOutput.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
 import io.trino.execution.Column;
 import io.trino.metadata.QualifiedObjectName;
 import io.trino.sql.analyzer.Analysis.SourceColumn;
-import org.testng.annotations.Test;
+import org.junit.jupiter.api.Test;
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestOutput.java:22]: This class still has TestNG method at L26.

**Comment by Damans227** [comment]
> Could you setup codestyle in your IDE? This is the documentation.
> https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#code-style

I am pretty sure I have setup Airlift code style in my IDE. Not sure why it didn't take effect. I will re-try.

**Comment by ebyhr** [comment]
@Damans227  Then, I would recommend to make sure "Optimize imports" setting.

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java [hunk 4]
```diff
@@ -16,14 +16,14 @@
...
-import org.testng.annotations.Test;
+import org.junit.jupiter.api.Test;
 
 import java.util.Optional;
 
 import static io.trino.spi.type.BigintType.BIGINT;
-import static org.testng.Assert.assertEquals;
-import static org.testng.Assert.assertFalse;
-import static org.testng.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (ebyhr) [inline core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java:24]: The argument order of `assertEquals` is different between TestNG and JUnit. Please see #9362 for the reference.

**Review by ebyhr — COMMENTED** [review]
Please use "rebase" (not merge) to follow upstream master branch.

File: core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java (hunk 4 above)
Comment (Damans227) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java:24]: @ebyhr  I see, so in that case I should replace the order of the arguments in all the **assertEquals** statements ? for eg:
 
 Change: 

`assertEquals(outer.tryResolveField(c1).get().getField(), outerColumn1);`

To:

`assertEquals(outerColumn1, outer.tryResolveField(c1).get().getField() );`

?

Comment (ebyhr) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java:24]: Yes, we should replace all of them.

Comment (Damans227) [reply core/trino-main/src/test/java/io/trino/sql/analyzer/TestScope.java:24]: Got it, and thanks for all the help. Appreciate it!

**Comment by Damans227** [comment]
Closing this and other PRs related to JUnit migration for now.

**Pull request closed without merging**